R/type_abline.R

```diff
@@ -0,0 +1,72 @@
+#' Add straight lines to a plot
+#'
+#' @inheritParams graphics::abline
+#' @inheritParams tinyplot
+#' @examples
+#' mod = lm(mpg ~ hp, data = mtcars)
+#' y = mtcars$mpg
+#' yhat = predict(mod)
+#' tinyplot(y, yhat, xlim = c(0, 40), ylim = c(0, 40))
+#' tinyplot_add(type = type_abline(a = 0, b = 1))
```

Comment:
This is great. Just so I know: What happens if you can `tinyplot(..., type = type_abline()` directly (i.e., without adding to an existing plot)? Do we need to add a `stop()` clause or warning for users that try to do this?

Reply:
good catch. I added a `stop()` logic and some tests